apis/v1beta1/validation/gateway.go

```diff
@@ -118,3 +121,39 @@ func ValidateTLSCertificateRefs(listeners []gatewayv1b1.Listener, path *field.Pa
 	}
 	return errs
 }
+
+// ValidateListenerNames validates the names of the listeners
+// must be unique within the Gateway
+func ValidateListenerNames(listeners []gatewayv1b1.Listener, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	nameMap := map[gatewayv1b1.SectionName]struct{}{}
+	for i, c := range listeners {
+		if _, found := nameMap[c.Name]; found {
+			errs = append(errs, field.Forbidden(path.Index(i).Child("name"), fmt.Sprintln("must be unique within the Gateway")))
```

Comment:
you want field.Duplicate, I think

Reply:
Changed this in https://github.com/kubernetes-sigs/gateway-api/pull/1933 :)